### PR TITLE
Add a logging hook for telemetry

### PR DESCRIPTION
**Pull request by tdb-alcorn**
I'm putting this up here to see if there is any interest in merging it. If so, I'll go through and fix up the imports to point to this repo again.

### CODE REVIEW

**Comment by pavelnikolov** [comment]
What problem are you trying to solve? Can you give a more specific example, please?

**Comment by tdb-alcorn** [comment]
This PR does three things:

1. A bug fix around hanging validation
2. Enable unmarshaling into pointer values as long as they point to a value of the correct type (I consider this a bug fix, it's clearly something a graphql library should do)
3. Add logging to the validator to enable telemetry e.g. logging the query name (we use this to profile our API by query type)

File: internal/exec/packer/packer.go [hunk 1]
```diff
@@ -354,8 +354,21 @@ func unmarshalInput(typ reflect.Type, input interface{}) (interface{}, error) {
...
+		value := reflect.ValueOf(input)
+		if value.CanAddr() {
+			return value.Addr().Interface(), nil
+		}
+		newValuePtr := reflect.New(typ.Elem())
+		newValue := reflect.Indirect(newValuePtr)
+		if newValue.CanSet() {
+			newValue.Set(value)
+			return newValuePtr.Interface(), nil
+		}
```
Comment (pavelnikolov) [inline internal/exec/packer/packer.go]: This is unrelated change.

**Comment by pavelnikolov** [comment]
Thanks you. Could you, please, create issues for these problems first and after we discuss the issues you can send one PR for each of the issues. IMHO, you should not try to solve multiple problems in a single PR. I'm closing this for now.
